Change to group messaging API in latest toxcore

# github.com/organ/golibtox
.gvm/pkgsets/go1.3/global/src/github.com/organ/golibtox/golibtox.go:797: not enough arguments in call to _Cfunc_tox_join_groupchat
.gvm/pkgsets/go1.3/global/src/github.com/organ/golibtox/golibtox.go:816: cannot use C.uint32_t(len(message)) (type C.uint32_t) as type C.uint16_t in argument to _Cfunc_tox_group_message_send
.gvm/pkgsets/go1.3/global/src/github.com/organ/golibtox/golibtox.go:835: cannot use C.uint32_t(len(action)) (type C.uint32_t) as type C.uint16_t in argument to _Cfunc_tox_group_action_send

# github.com/nikwen/golibtox
../../.gvm/pkgsets/go1.3/global/src/github.com/nikwen/golibtox/golibtox.go: In function ‘set_callback_group_invite’:
../../.gvm/pkgsets/go1.3/global/src/github.com/nikwen/golibtox/golibtox.go:44:1: warning: passing argument 2 of ‘tox_callback_group_invite’ from incompatible pointer type [enabled by default]
 HOOK(callback_group_invite)
 ^
In file included from ../../.gvm/pkgsets/go1.3/global/src/github.com/nikwen/golibtox/golibtox.go:8:0:
/usr/local/include/tox/tox.h:426:6: note: expected ‘void (*)(struct Tox *, int32_t,  const uint8_t *, uint16_t,  void *)’ but argument is of type ‘void (*)(struct Tox *, int32_t,  uint8_t,  const uint8_t *, uint16_t,  void *)’
 void tox_callback_group_invite(Tox *tox, void (*function)(Tox *tox, int32_t, const uint8_t *, uint16_t, void *),
      ^
# github.com/nikwen/golibtox
../../.gvm/pkgsets/go1.3/global/src/github.com/nikwen/golibtox/golibtox.go:891: cannot use (*C.int32_t)(unsafe.Pointer(&cchatlist[0])) (type *C.int32_t) as type *C.int in argument to _Cfunc_tox_get_chatlist

--- a/golibtox.go
+++ b/golibtox.go
@@ -24,7 +24,7 @@ void hook_callback_user_status(Tox*, int32_t, uint8_t, void*);
 void hook_callback_typing_change(Tox*, int32_t, uint8_t, void*);
 void hook_callback_read_receipt(Tox*, int32_t, uint32_t, void*);
 void hook_callback_connection_status(Tox*, int32_t, uint8_t, void*);
-void hook_callback_group_invite(Tox*, int32_t, const uint8_t*, void*);
+void hook_callback_group_invite(Tox*, int32_t, uint8_t, const uint8_t*, uint16_t, void*);
 void hook_callback_group_message(Tox*, int, int, const uint8_t*, uint16_t, void*);
 void hook_callback_group_action(Tox*, int, int, const uint8_t*, uint16_t, void*);
 void hook_callback_group_namelist_change(Tox*, int, int, uint8_t, void*);
@@ -87,7 +87,7 @@ type OnReadReceipt func(tox *Tox, friendnumber int32, receipt uint32)
 type OnConnectionStatus func(tox *Tox, friendnumber int32, online bool)
 
 // OnGroupInvite is a callback function called by Tox when receiving a Groupchat invite.
-type OnGroupInvite func(tox *Tox, friendnumber int32, groupPublicKey []byte)
+type OnGroupInvite func(tox *Tox, friendnumber int32, _type uint8, groupPublicKey []byte, length uint16)
 
 // OnGroupMessage is a callback function called by Tox when receiving a Groupchat message.
 type OnGroupMessage func(tox *Tox, groupnumber int, friendgroupnumber int, message []byte, length uint16)
@@ -794,7 +794,7 @@ func (t *Tox) JoinGroupchat(friendnumber int32, friendGroupPublicKey []byte) (in
 		return -1, ErrArgs
 	}
 
-	ret := C.tox_join_groupchat(t.tox, (C.int32_t)(friendnumber), (*C.uint8_t)(&friendGroupPublicKey[0]))
+	ret := C.tox_join_groupchat(t.tox, (C.int32_t)(friendnumber), (*C.uint8_t)(&friendGroupPublicKey[0]), (C.uint16_t)(friendGroupPublicKey[1]))
 
 	if ret == -1 {
 		return -1, ErrFuncFail
@@ -813,7 +813,7 @@ func (t *Tox) GroupMessageSend(groupnumber int, message []byte) error {
 		return ErrArgs
 	}
 
-	ret := C.tox_group_message_send(t.tox, (C.int)(groupnumber), (*C.uint8_t)(&message[0]), (C.uint32_t)(len(message)))
+	ret := C.tox_group_message_send(t.tox, (C.int)(groupnumber), (*C.uint8_t)(&message[0]), (C.uint16_t)(len(message)))
 
 	if ret == -1 {
 		return ErrFuncFail
@@ -832,7 +832,7 @@ func (t *Tox) GroupActionSend(groupnumber int, action []byte) error {
 		return ErrArgs
 	}
 
-	ret := C.tox_group_action_send(t.tox, (C.int)(groupnumber), (*C.uint8_t)(&action[0]), (C.uint32_t)(len(action)))
+	ret := C.tox_group_action_send(t.tox, (C.int)(groupnumber), (*C.uint8_t)(&action[0]), (C.uint16_t)(len(action)))
 
 	if ret == -1 {
 		return ErrFuncFail
@@ -888,7 +888,7 @@ func (t *Tox) GetChatlist() ([]int, error) {
 	size, _ := t.CountChatlist()
 	cchatlist := make([]int, size)
 
-	n := C.tox_get_chatlist(t.tox, (*C.int)(unsafe.Pointer(&cchatlist[0])), (C.uint32_t)(size))
+	n := C.tox_get_chatlist(t.tox, (*C.int32_t)(unsafe.Pointer(&cchatlist[0])), (C.uint32_t)(size))
 
 	chatlist := cchatlist[:n]
 

--- a/hooks.go
+++ b/hooks.go
@@ -60,8 +60,8 @@ func hook_callback_connection_status(t unsafe.Pointer, friendnumber C.int32_t, c
 }
 
 //export hook_callback_group_invite
-func hook_callback_group_invite(t unsafe.Pointer, friendnumber C.int32_t, groupPublicKey *C.uint8_t, tox unsafe.Pointer) {
-	(*Tox)(tox).onGroupInvite((*Tox)(tox), int32(friendnumber), C.GoBytes((unsafe.Pointer)(groupPublicKey), CLIENT_ID_SIZE))
+func hook_callback_group_invite(t unsafe.Pointer, friendnumber C.int32_t, _type C.uint8_t, groupPublicKey *C.uint8_t, length C.uint16_t, tox unsafe.Pointer) {
+	(*Tox)(tox).onGroupInvite((*Tox)(tox), int32(friendnumber), uint8(_type), C.GoBytes((unsafe.Pointer)(groupPublicKey), (C.int)(length)), uint16(length))
 }
 
 //export hook_callback_group_message

